File: tests/test_code/java/FieldSensitivityTest.java
package com.example.test;

import javax.servlet.http.HttpServletRequest;
import java.sql.Connection;
import java.sql.Statement;

public class FieldSensitivityTest {
    
    class UserDTO {
        public String name;
        public int id;
    }

    public void test(HttpServletRequest request, Connection conn) throws Exception {
        UserDTO user = new UserDTO();
        user.name = request.getParameter("name");
        user.id = 123; // Safe constant
        
        Statement stmt = conn.createStatement();

        // VULNERABLE: user.name is tainted
        stmt.executeQuery("SELECT * FROM users WHERE name = '" + user.name + "'");

        // SAFE: user.id is clean
        stmt.executeQuery("SELECT * FROM users WHERE id = " + user.id);
    }
}

File: tests/test_code/python/field_sensitivity_test.py
from flask import Flask, request
import sqlite3

app = Flask(__name__)

class User:
    def __init__(self):
        self.name = ""
        self.id = 0

@app.route('/test')
def test_field_sensitivity():
    user = User()
    user.name = request.args.get('name')
    user.id = 123 # Safe constant
    
    conn = sqlite3.connect('db.db')
    cursor = conn.cursor()

    # VULNERABLE: user.name is tainted
    cursor.execute("SELECT * FROM users WHERE name = '" + user.name + "'")

    # SAFE: user.id is a constant, should be ignored
    cursor.execute("SELECT * FROM users WHERE id = " + str(user.id))

    return "Done"
